### DISCUSSION TITLE
[Feature enhancement request] useReducer support?

### DISCUSSION THREAD

**Discussion by shoes22** [Aminadav/react-useStateRef#9] [Ideas]
First off, great idea for a library. I am looking for something like this because I'm working on a project that uses react-native-video, and we need to store the current playback position of the media file in state (which updates several times a second), which a lot of other functions end up referencing. 

Because of this requirement, a lot of the useCallback functions we have end up needing that playback position as a dependency, which means those functions must get re-calculated on every single render, tanking the performance of our app. 

Your solution seems virtually ideal to solve the problem we have, with one minor issue. To (marginally) help improve the performance of the component in question, we converted all the useState calls to a single useReducer and dispatch functions. 

Would it be difficult to adapt this code so it can work with useReducer as well?

**Comment by Aminadav** [comment]
If it was implemented can you provide an example code how it will be used?
How your code looks today, and how it will look with `useReducerRef`?

Please keep the examples for the minimal code with only the relevant parts for us to understand.

**Comment by Aminadav** [comment]
I'm closing the discussion let's move it to an issue instead.
